src/pymedphys/dcm/_level2/dcmstruct.py

```diff
@@ -32,14 +32,72 @@
 from ...geometry import (
     cubify_cube_definition, cube_vertices, cube_vectors)
 
-from .._level1.dcmdose import (
-    pull_structure, contour_to_points)
-
 IMPORTS = get_imports(globals())
 
 
 # pylint: disable=C0103
 
+def pull_structure_by_number(number, dcm_struct):
+    contours_by_slice_raw = [
+        item.ContourData
+        for item in dcm_struct.ROIContourSequence[number].ContourSequence
+    ]
+    x = [
+        np.array(item[0::3])
+        for item in contours_by_slice_raw]
+    y = [
+        np.array(item[1::3])
+        for item in contours_by_slice_raw]
+    z = [
+        np.array(item[2::3])
+        for item in contours_by_slice_raw]
+
+    return x, y, z
+
+
+def pull_structure(string, dcm_struct):
```

Comment:
I shifted structure logic into the level 2 structure dicom file, it made more sense there anyway. That freed me up to have dicom dose depend on the structure file instead of the other way round. If I didn't move that logic I would have had to make a level 5 as well as level 4. We are free to make as many levels as needed, but often some simple refactoring can lessen the number of levels.

This would also probably be why the tests are failing. In my haste between packing I didn't actually transfer any of the imports across between the files. Oops.

Reply:
Gotcha.